### PR TITLE
[HotFix] Disable HIP iGEMM V6R1 DLOps Kernels

### PR DESCRIPTION
**Pull request by junliume**
The actual issue:

These commands will get stuck while executing on MI200s:
```
root@hpe-sjc2-43:/opt/rocm# ./bin/MIOpenDriver conv -n 256 -c 256 -H 2 -W 2 -k 512 -y 1 -x 1 -p 0 -q 0 -u 2 -v 2 -l 1 -j 1 -m conv -g 1 -F 1 -t 1
MIOpenDriver conv -n 256 -c 256 -H 2 -W 2 -k 512 -y 1 -x 1 -p 0 -q 0 -u 2 -v 2 -l 1 -j 1 -m conv -g 1 -F 1 -t 1
^C


root@hpe-sjc2-43:/opt/rocm# ./bin/MIOpenDriver conv -n 512 -c 256 -H 2 -W 2 -k 512 -y 1 -x 1 -p 0 -q 0 -u 2 -v 2 -l 1 -j 1 -m conv -g 1 -F 1 -t 1
MIOpenDriver conv -n 512 -c 256 -H 2 -W 2 -k 512 -y 1 -x 1 -p 0 -q 0 -u 2 -v 2 -l 1 -j 1 -m conv -g 1 -F 1 -t 1
Memory access fault by GPU node-8 (Agent handle: 0x2430240) on address 0x7f71c7e18000. Reason: Unknown.
Aborted (core dumped)
```

But for any other batch sizes it will pass:
```
root@hpe-sjc2-43:/opt/rocm# ./bin/MIOpenDriver conv -n 522 -c 256 -H 2 -W 2 -k 512 -y 1 -x 1 -p 0 -q 0 -u 2 -v 2 -l 1 -j 1 -m conv -g 1 -F 1 -t 1
MIOpenDriver conv -n 522 -c 256 -H 2 -W 2 -k 512 -y 1 -x 1 -p 0 -q 0 -u 2 -v 2 -l 1 -j 1 -m conv -g 1 -F 1 -t 1
MIOpen Forward Conv. Algorithm: 5, Solution: 107/ConvAsmImplicitGemmGTCDynamicFwdXdlopsNHWC
GPU Kernel Time Forward Conv. Elapsed: 0.043342 ms (average)
stats: name, n, c, ho, wo, x, y, k, flopCnt, bytesRead, bytesWritten, GFLOPs, GB/s, timeMs
stats: fwd-conv1x1u2, 522, 256, 1, 1, 1, 1, 512,  136839168, 2662400, 1069056, 3157, 86, 0.043342
Forward Convolution Verifies OK on CPU reference (3.71208e-08)
```

It is found that in these cases, `MIOPEN_DEBUG_CONV_CK_IGEMM_FWD_V6R1_DLOPS_NCHW=0` will let the configurations pass.

HIP iGEMM V6R1 DLOps Kernels are not using XDLOps and should be less efficient than ASM iGEMM XDLOps kernels.

### CODE REVIEW

File: src/solver/conv_ck_igemm_fwd_v6r1_dlops_nchw.cpp [hunk 1]
```diff
@@ -83,7 +85,7 @@ bool PerformanceConvCkIgemmFwdV6r1DlopsNchw::IsValid(const ProblemDescription& p
 bool ConvCkIgemmFwdV6r1DlopsNchw::IsApplicable(const ConvolutionContext& ctx,
                                                const ProblemDescription& problem) const
 {
-    if(miopen::IsDisabled(MIOPEN_DEBUG_CONV_CK_IGEMM_FWD_V6R1_DLOPS_NCHW{}))
+    if(miopen::IsDisabled(MIOPEN_DEBUG_CONV_CK_IGEMM_FWD_V6R1_DLOPS_NCHW{}) || WORKAROUND_SWDEV_411729)
```
Comment (atamazov) [inline src/solver/conv_ck_igemm_fwd_v6r1_dlops_nchw.cpp]: Let's allow overriding the W/A with env var
```suggestion
#if WORKAROUND_SWDEV_411729
    if(!miopen::IsEnabled(MIOPEN_DEBUG_CONV_CK_IGEMM_FWD_V6R1_DLOPS_NCHW{}))
#endif
    if(miopen::IsDisabled(MIOPEN_DEBUG_CONV_CK_IGEMM_FWD_V6R1_DLOPS_NCHW{}))
#endif
```

**Review by atamazov — CHANGES_REQUESTED** [review]

**Comment by atamazov** [comment]
@junliume Shall we open a ticket and assign an engineer to investigate the problem and find root reason. I can do that for you, just let me know.

**Comment by junliume** [comment]
> @junliume Shall we open a ticket and assign an engineer to investigate the problem and find root reason. I can do that for you, just let me know.

@atamazov  thank you! and only if you have some time then. This kernel is not using XDLOps and digging into the root cause might be time consuming, I would suggest take a look while not spending too much time on it.

**Comment by atamazov** [comment]
@junliume 

> ...
> But for any other batch sizes it will pass:
> 
> ```
> ...
> MIOpen Forward Conv. Algorithm: 5, Solution: 107/ConvAsmImplicitGemmGTCDynamicFwdXdlopsNHWC
> ```

As you can see, in this example MIOpen selects another solver, ConvAsmImplicitGemmGTCDynamicFwdXdlopsNHWC (I guess because of HIP iGEMM V6R1 DLOps is not applicable) and that if why the test is passing.

**Comment by junliume** [comment]
@atamazov Yes! For some reason, when the mini batch size is in the form `n == 2^m`, the find algorithm thinks the HIP iGEMM kernels are faster, otherwise the ASM iGEMM will kick in.
I assume actually in all cases ASM should kick in instead.

**Comment by atamazov** [comment]
@junliume 
> thank you! and only if you have some time then. 

I mean only opening a ticket and proposing some assignment ;) Because I do not have MI200 on hand. BTW `ConvCkIgemmFwdV6r1DlopsNchw` is in the list of deprecated solvers (#2080), so we can simply disable it ~for MI200~ as a W/A and forget about it...? @JehandadKhan Do you agree?

**Comment by junliume** [comment]
> @junliume
> 
> > thank you! and only if you have some time then.
> 
> I mean only opening a ticket and proposing some assignment ;) Because I do not have MI200 on hand. BTW `ConvCkIgemmFwdV6r1DlopsNchw` is in the list of deprecated solvers, so we can simply disable it for MI200 as a W/A and forget about it...? @JehandadKhan Do you agree?

@atamazov actually, we can reproduce this issue on Navi systems, since the kernel itself does not utilize XDLOps.

**Comment by atamazov** [comment]
@junliume Please add
```bash
# MIOPEN_DEBUG_CONV_CK_IGEMM_FWD_V6R1_DLOPS_NCHW=1 is due to WORKAROUND_SWDEV_411729
...
MIOPEN_DEBUG_CONV_CK_IGEMM_FWD_V6R1_DLOPS_NCHW=1
```
to `smoke_solver_ConvCkIgemmFwdV6r1DlopsNchw` in test\CMakeLists.txt or it will fail.

**Comment by atamazov** [comment]
@junliume 
> actually, we can reproduce this issue on Navi systems, since the kernel itself does not utilize XDLOps.

Oh, I see. Let's check that the reason of the issue is NOT the recent compiler changes and then we can forget about it (well... er... open some low-urgency ticket ;). Otherwise everything is in danger and it seems reasonable to escalate the issue to the compiler team.

File: test/CMakeLists.txt [hunk 2]
```diff
@@ -1756,7 +1756,8 @@ COMMAND ${IMPLICITGEMM_TESTING_ENV} MIOPEN_LOG_LEVEL=5 $<TARGET_FILE:test_conv2d
 
 set(CONV_CK_IGEMM_FWD_V6R1_DLOPS_NCHW_ENV
```
Comment (atamazov) [inline test/CMakeLists.txt:1758]: [Recommendation] Can you please add comment from https://github.com/ROCmSoftwarePlatform/MIOpen/pull/2306#issuecomment-1668655231 to explain why we use env var to enable the solver.

File: test/CMakeLists.txt [hunk 3]
```diff
@@ -2065,7 +2066,7 @@ add_custom_test(smoke_solver_ConvAsmImplicitGemmGTCDynamicFwdDlopsNCHWC GFX900_D
 add_custom_test(smoke_solver_ConvCkIgemmFwdV6r1DlopsNchw GFX103X_ENABLED HALF_ENABLED TEST_TUNING
     COMMAND MIOPEN_FIND_ENFORCE=SEARCH_DB_UPDATE MIOPEN_DEBUG_TUNING_ITERATIONS_MAX=2
       MIOPEN_DEBUG_CONVOLUTION_ATTRIB_FP16_ALT_IMPL=0
-      MIOPEN_FIND_MODE=normal MIOPEN_DEBUG_FIND_ONLY_SOLVER=ConvCkIgemmFwdV6r1DlopsNchw $<TARGET_FILE:test_conv2d>
+      MIOPEN_FIND_MODE=normal MIOPEN_DEBUG_FIND_ONLY_SOLVER=ConvCkIgemmFwdV6r1DlopsNchw MIOPEN_DEBUG_CONV_CK_IGEMM_FWD_V6R1_DLOPS_NCHW=1 $<TARGET_FILE:test_conv2d>
```
Comment (atamazov) [inline test/CMakeLists.txt:2071]: Ditto.

**Review by atamazov — APPROVED** [review]
LGTM

File: test/CMakeLists.txt (hunk 2 above)
Comment (junliume) [reply test/CMakeLists.txt:1758]: Done. Next time I will do them all at once instead of separate commits ;-P

**Review by zjing14 — APPROVED** [review]

**Pull request merged**